sdk/identity/identity/src/credentials/vscodeCredential.ts

```diff
@@ -19,12 +22,44 @@ const AzureAccountClientId = "aebc6443-996d-45c2-90f0-388ff96faa56"; // VSC: 'ae
 const VSCodeUserName = "VS Code Azure";
 const logger = credentialLogger("VSCodeCredential");
 
+/**
+ * Attempts to load the tenant from the VSCode configurations of the current OS.
+ * If it fails at any point, returns undefined.
+ */
+function getTenantIdFromVSCode(): string | undefined {
```

Comment:
This method doesn't consider alternative paths for the settings.json, and it doesn't consider multiple settings.json files.

For context:
VSCode does allow multiple settings.json files, depending on the used extensions, as well as to customize where this file is depending on a multitude of parameters. The logic of how this happen can be seen around this file: https://github.com/microsoft/vscode/blob/master/src/vs/platform/workspace/common/workspace.ts#L177-L179 and here is how the extension vscode-azure-account is using those tools: https://github.com/microsoft/vscode-azure-account/blob/master/src/azure-account.ts#L203 (the link only shows somewhere in which the VSCode source code is being used to access the `azure.tenant` property).

We will stick to the most simple approach for now, since we can't take a dependency on VSCode, and since the users can specify the their tenants through the parameters.